### ISSUE TITLE
Failed to create-react-native-app

### ISSUE THREAD

**Issue by ngluva** [react-community/create-react-native-app#687]
## Description

Have been trying for several days to use create-react-native-app.  
Initially I had problems with "The filename, directory name, or volume label syntax is incorrect.", but I resolved this error by removing node and installing it to "C:" drive

I still cannot use "create-react-native-app" but I do get further into the creation of an app.

## Expected Behavior
 Creation of an App template

What action did you perform, and what did you expect to happen?
create-react-native-app myapp 
## Observed Behavior
create-react-native-app  Failed to install

What actually happened when you performed the above actions?
"Failed to install"

PS C:\max> create-react-native-app myapp
Creating a new React Native app in H:\max\myapp.

Using package manager as yarnpkg with yarn interface.
Installing packages. This might take a couple minutes.
Installing react-native-scripts...
```
yarn add v1.7.0
info No lockfile found.
[1/4] Resolving packages...
warning react-native-scripts > xdl > auth0 > rest-facade > superagent-proxy > proxy-agent > socks-proxy-agent > socks@1.1.10: If using 2.x branch, please upgrade to at least 2.1.6 to avoid a serious bug with socket data flow and an import issue introduced in 2.1.0
[2/4] Fetching packages...
info @expo/ngrok-bin-darwin-x64@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-darwin-x64@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-darwin-ia32@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-darwin-ia32@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-darwin-ia32@2.2.8: The CPU architecture "x64" is incompatible with this module.
info @expo/ngrok-bin-freebsd-ia32@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-freebsd-ia32@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-freebsd-ia32@2.2.8: The CPU architecture "x64" is incompatible with this module.
info @expo/ngrok-bin-linux-arm64@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-linux-arm64@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-linux-arm64@2.2.8: The CPU architecture "x64" is incompatible with this module.
info @expo/ngrok-bin-freebsd-x64@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-freebsd-x64@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-linux-ia32@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-linux-ia32@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-linux-ia32@2.2.8: The CPU architecture "x64" is incompatible with this module.
info @expo/ngrok-bin-linux-x64@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-linux-x64@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-sunos-x64@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-sunos-x64@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-win32-ia32@2.2.8-beta.1: The CPU architecture "x64" is incompatible with this module.
info "@expo/ngrok-bin-win32-ia32@2.2.8-beta.1" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-linux-arm@2.2.8: The platform "win32" is incompatible with this module.
info "@expo/ngrok-bin-linux-arm@2.2.8" is an optional dependency and failed compatibility check. Excluding it from installation.
info @expo/ngrok-bin-linux-arm@2.2.8: The CPU architecture "x64" is incompatible with this module.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 323 new dependencies.
info Direct dependencies
└─ react-native-scripts@1.14.0
info All dependencies
├─ @expo/bunyan@1.8.10
├─ @expo/json-file@5.3.0
├─ @expo/ngrok-bin-win32-x64@2.2.8-beta.1
├─ @expo/ngrok-bin@2.2.8-beta.3
├─ @expo/ngrok@2.4.2
├─ @expo/osascript@1.8.1
├─ @expo/schemer@1.1.0
├─ @segment/loosely-validate-event@1.1.2
├─ @types/events@1.2.0
├─ @types/glob@5.0.35
├─ @types/minimatch@3.0.3
├─ @types/node@10.3.4
├─ accepts@1.3.5
├─ ajv@5.5.2
├─ analytics-node@2.4.1
├─ ansi-regex@3.0.0
├─ ansi-styles@3.2.1
├─ array-flatten@1.1.1
├─ array-union@1.0.2
├─ array-uniq@1.0.3
├─ asap@2.0.6
├─ asn1@0.2.3
├─ ast-types@0.11.5
├─ async@0.9.2
├─ asynckit@0.4.0
├─ auth0-js@9.3.3
├─ auth0@2.9.1
├─ aws-sign2@0.7.0
├─ aws4@1.7.0
├─ axios@0.16.2
├─ babel-plugin-syntax-flow@6.18.0
├─ babel-plugin-transform-flow-strip-types@6.22.0
├─ babel-polyfill@6.26.0
├─ babel-preset-flow@6.23.0
├─ babel-runtime@6.26.0
├─ balanced-match@1.0.0
├─ bcrypt-pbkdf@1.0.1
├─ body-parser@1.18.3
├─ brace-expansion@1.1.11
├─ buffer-equal-constant-time@1.0.1
├─ buffer-from@1.1.0
├─ callsite@1.0.0
├─ capture-stack-trace@1.0.0
├─ caseless@0.12.0
├─ change-case@2.3.1
├─ chardet@0.4.2
├─ charenc@0.0.2
├─ chownr@1.0.1
├─ clone@2.1.1
├─ color-convert@1.9.2
├─ color-name@1.1.1
├─ combined-stream@1.0.6
├─ commander@2.15.1
├─ component-emitter@1.2.1
├─ component-type@1.2.1
├─ concat-map@0.0.1
├─ concat-stream@1.6.2
├─ constant-case@1.1.2
├─ content-disposition@0.5.2
├─ cookie-signature@1.0.6
├─ cookiejar@2.1.2
├─ core-js@2.5.7
├─ create-error-class@3.0.2
├─ cross-spawn@5.1.0
├─ crypt@0.0.2
├─ crypto-js@3.1.9-1
├─ crypto-token@1.0.1
├─ dashdash@1.14.1
├─ data-uri-to-buffer@1.2.0
├─ decache@4.4.0
├─ deep-diff@0.3.4
├─ deep-is@0.1.3
├─ deepmerge@1.5.2
├─ degenerator@1.0.4
├─ delay-async@1.1.0
├─ delayed-stream@1.0.0
├─ destroy@1.0.4
├─ dot-case@1.1.2
├─ duplexer3@0.1.4
├─ ecc-jsbn@0.1.1
├─ ecdsa-sig-formatter@1.0.10
├─ ee-first@1.1.1
├─ encoding@0.1.12
├─ es-abstract@1.12.0
├─ es-to-primitive@1.1.1
├─ es6-promise@4.2.4
├─ es6-promisify@5.0.0
├─ escodegen@1.10.0
├─ esprima@3.1.3
├─ estraverse@4.2.0
├─ esutils@2.0.2
├─ exec-async@2.2.0
├─ exists-async@2.0.0
├─ express@4.16.3
├─ extend@3.0.1
├─ external-editor@2.2.0
├─ extsprintf@1.3.0
├─ fast-deep-equal@1.1.0
├─ fast-json-stable-stringify@2.0.0
├─ fast-levenshtein@2.0.6
├─ fbjs@0.8.17
├─ file-type@4.4.0
├─ file-uri-to-path@1.0.0
├─ finalhandler@1.1.1
├─ foreach@2.0.5
├─ forever-agent@0.6.1
├─ form-data@2.3.2
├─ formidable@1.2.1
├─ forwarded@0.1.2
├─ freeport-async@1.1.1
├─ fs-extra@3.0.1
├─ fs-minipass@1.2.5
├─ fs.realpath@1.0.0
├─ ftp@0.3.10
├─ get-stream@3.0.0
├─ get-uri@2.0.2
├─ getenv@0.7.0
├─ getpass@0.1.7
├─ glob-promise@3.4.0
├─ globby@6.1.0
├─ got@6.7.1
├─ har-schema@2.0.0
├─ har-validator@5.0.3
├─ has-flag@3.0.0
├─ has-symbols@1.0.0
├─ hasbin@1.2.3
├─ hoist-non-react-statics@2.5.5
├─ home-dir@1.0.0
├─ http-signature@1.2.0
├─ iconv-lite@0.4.23
├─ idtoken-verifier@1.2.0
├─ idx@2.4.0
├─ indent-string@3.2.0
├─ inflight@1.0.6
├─ inherits@2.0.3
├─ inquirer@3.3.0
├─ instapromise@2.0.7
├─ invariant@2.2.4
├─ ip@1.1.5
├─ ipaddr.js@1.6.0
├─ is-buffer@1.1.6
├─ is-callable@1.1.3
├─ is-date-object@1.0.1
├─ is-fullwidth-code-point@2.0.0
├─ is-lower-case@1.1.3
├─ is-promise@2.1.0
├─ is-redirect@1.0.0
├─ is-regex@1.0.4
├─ is-retry-allowed@1.1.0
├─ is-stream@1.1.0
├─ is-symbol@1.0.1
├─ is-typedarray@1.0.0
├─ is-upper-case@1.1.2
├─ isarray@1.0.0
├─ isemail@2.2.1
├─ isexe@2.0.0
├─ isomorphic-fetch@2.2.1
├─ isstream@0.1.2
├─ items@2.1.1
├─ joi@10.6.0
├─ join-component@1.1.0
├─ js-tokens@3.0.2
├─ json-schema-traverse@0.3.1
├─ json-schema@0.2.3
├─ json-stringify-safe@5.0.1
├─ json5@0.5.1
├─ jsonfile@2.4.0
├─ jsonschema@1.2.4
├─ jsonwebtoken@7.4.3
├─ jsprim@1.4.1
├─ jwa@1.1.6
├─ jws@3.1.5
├─ levn@0.3.0
├─ lock@0.1.4
├─ lodash-es@4.17.10
├─ lodash.once@4.1.1
├─ lodash@4.17.10
├─ logfmt@1.2.1
├─ lower-case-first@1.0.2
├─ lower-case@1.1.4
├─ lowercase-keys@1.0.1
├─ lru-cache@4.1.3
├─ lru-memoizer@1.12.0
├─ match-require@2.1.0
├─ md5@2.2.1
├─ md5hex@1.0.0
├─ media-typer@0.3.0
├─ merge-descriptors@1.0.1
├─ methods@1.1.2
├─ mime-db@1.33.0
├─ mime@1.4.1
├─ mimic-fn@1.2.0
├─ minimatch@3.0.4
├─ minimist@1.2.0
├─ minizlib@1.1.0
├─ mkdirp-promise@5.0.1
├─ mkdirp@0.5.1
├─ moment@2.22.2
├─ mv@2.1.1
├─ ncp@2.0.0
├─ negotiator@0.6.1
├─ netmask@1.0.6
├─ next-tick@1.0.0
├─ node-fetch@1.7.3
├─ oauth-sign@0.8.2
├─ object-assign@4.1.1
├─ object-keys@1.0.12
├─ object.assign@4.1.0
├─ object.getownpropertydescriptors@2.0.3
├─ onetime@2.0.1
├─ opn@4.0.2
├─ optionator@0.8.2
├─ os-tmpdir@1.0.2
├─ pac-proxy-agent@2.0.2
├─ pac-resolver@3.0.0
├─ param-case@1.1.2
├─ pascal-case@1.1.2
├─ path-case@1.1.2
├─ path-exists@3.0.0
├─ path-is-absolute@1.0.1
├─ path-to-regexp@0.1.7
├─ performance-now@2.1.0
├─ pify@2.3.0
├─ pinkie@2.0.4
├─ plist@2.1.0
├─ prepend-http@1.0.4
├─ probe-image-size@3.2.0
├─ process-nextick-args@2.0.0
├─ progress@2.0.0
├─ promise@7.3.1
├─ proxy-addr@2.0.3
├─ proxy-agent@2.3.1
├─ proxy-from-env@1.0.0
├─ pseudomap@1.0.2
├─ psl@1.1.28
├─ qrcode-terminal@0.11.0
├─ qs@6.5.2
├─ querystring@0.2.0
├─ raven-js@3.26.2
├─ raven@2.6.2
├─ raw-body@2.3.3
├─ react-native-scripts@1.14.0
├─ react-redux@5.0.7
├─ react@16.4.1
├─ readable-stream@2.3.6
├─ redux-logger@2.10.2
├─ redux@3.7.2
├─ regenerator-runtime@0.11.1
├─ remove-trailing-slash@0.1.0
├─ replace-string@1.1.0
├─ request-progress@3.0.0
├─ request-promise-core@1.1.1
├─ request-promise-native@1.0.5
├─ request@2.87.0
├─ rest-facade@1.10.1
├─ restore-cursor@2.0.0
├─ retry@0.10.1
├─ rimraf@2.6.2
├─ rx-lite-aggregates@4.0.8
├─ rx-lite@4.0.8
├─ rxjs@5.5.11
├─ safe-json-stringify@1.2.0
├─ safer-buffer@2.1.2
├─ semver@5.5.0
├─ serve-static@1.13.2
├─ setimmediate@1.0.5
├─ shebang-command@1.2.0
├─ shebang-regex@1.0.0
├─ signal-exit@3.0.2
├─ slugid@1.1.0
├─ slugify@1.3.0
├─ smart-buffer@1.1.15
├─ socks@1.1.10
├─ source-map-support@0.4.18
├─ source-map@0.5.7
├─ split@1.0.1
├─ sshpk@1.14.2
├─ stack-trace@0.0.10
├─ stealthy-require@1.1.1
├─ stream-parser@0.3.1
├─ string_decoder@1.1.1
├─ superagent-proxy@1.0.3
├─ superagent-retry@0.6.0
├─ supports-color@5.4.0
├─ swap-case@1.1.2
├─ symbol-observable@1.2.0
├─ tar@4.4.4
├─ thenify-all@1.6.0
├─ thenify@3.3.0
├─ throttleit@1.0.0
├─ thunkify@2.1.2
├─ timed-out@4.0.1
├─ title-case@1.1.2
├─ tmp@0.0.33
├─ topo@2.0.2
├─ tough-cookie@2.4.2
├─ tree-kill@1.2.0
├─ tunnel-agent@0.6.0
├─ tweetnacl@0.14.5
├─ type-is@1.6.16
├─ typedarray@0.0.6
├─ ua-parser-js@0.7.18
├─ unpipe@1.0.0
├─ unzip-response@2.0.1
├─ url-parse-lax@1.0.0
├─ url@0.11.0
├─ util-deprecate@1.0.2
├─ util.promisify@1.0.0
├─ utils-merge@1.0.1
├─ vary@1.1.2
├─ verror@1.10.0
├─ very-fast-args@1.1.0
├─ whatwg-fetch@2.0.4
├─ which@1.3.1
├─ winchan@0.2.0
├─ wordwrap@1.0.0
├─ xdl@48.1.4
├─ xmlbuilder@8.2.2
├─ xmldom@0.1.27
├─ xregexp@2.0.0
├─ xtend@4.0.1
├─ yallist@2.1.2
└─ yesno@0.0.1
Done in 43.89s.
Installing dependencies using yarn...

yarn install v1.7.0
[1/4] Resolving packages...
warning jest-expo > jest > jest-cli > istanbul-api > istanbul-lib-hook@1.2.1: 1.2.0 should have been a major version bump
[2/4] Fetching packages...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "http://registry.npmjs.org/compression/-/compression-1.7.2.tgz: socket hang up".
info If you think this is a bug, please open a bug report with the information provided in "C:\\max\myapp\\yarn-error.log
[yarn-error.log](https://github.com/react-community/create-react-native-app/files/2118200/yarn-error.log)
".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Failed to install

```
I have included the error log referred to above.
[yarn-error.log](https://github.com/react-community/create-react-native-app/files/2118224/yarn-error.log)
 ## Environment

Please run these commands in the project folder and fill in their results:
 
'npm ls react-native-scripts`:`-- (empty)
'npm ls react-native`: `-- (empty)
`npm ls expo`: `-- (empty)
`node -v`: v8.11.3
`npm -v`: 5.6.0
`yarn --version`: 1.7.0
`watchman version`: none
 
Also specify:
 
1. Operating system: windows 10 version 1803 (OS Build) 17134.112)
2. Phone/emulator/simulator & version: Samsung Galaxy J7-Pro
 
## Reproducible Demo
I have installed and created a App using a virtual computer (Windows 8 and Ubuntu) on my computer without any problems so you as you can imagine this road-block is causing me great consternation.

## Notice the error:
"An unexpected error occurred: "http://registry.npmjs.org/compression/-/compression-1.7.2.tgz: socket hang up". This happens every time I run the script on this computer regardless of the network or the ISP, except for when I use a virtual computer.

Can I do a forklift installation from my virtual computer?

**Comment by daviskoh** [comment]
Just had this prob. Mine was a problem w/ expo [requiring Node v >= 8](https://github.com/react-community/create-react-native-app/issues/687). u will still see those expo logs but as warnings, but should be able to install.

**Comment by ngluva** [comment]
Thanks for your post but I never got it to install. I installed it onto a virtual machine on my laptop and then copied the folder onto my laptop. Much of the install assume you have only one NIC instead of looking for the NIC with a access to the net.
After spending nearly a week learning about things that I would have thought fundamental to development of the script I stopped, because I then ran into a Red screen that I just couldn't get past. I have had the time to get back to this project as I have other things to do.
I think this is the best script to find the correct IP Address (windows anyway) because it looks up the routing table https://github.com/Aldaviva/local-ipv4-address .
